arangod/Aql/SortBlock.cpp

```diff
@@ -55,27 +59,30 @@ void stealRow(std::unordered_map<arangodb::aql::AqlValue, arangodb::aql::AqlValu
 
           if (vCount == 0) {
             // Was already stolen for another block
-            arangodb::aql::AqlValue copy = original.clone();
-            try {
+            AqlValue copy = original.clone();
+            Result res = catchVoidToResult([&cache, &original, &copy]() -> void {
               TRI_IF_FAILURE("SortBlock::doSortingCache") {
                 THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
               }
               cache.emplace(original, copy);
-            } catch (...) {
+            });
+            if (res.fail()) {
               copy.destroy();
-              throw;
+              THROW_ARANGO_EXCEPTION(res);
             }
 
-            try {
+            res = catchVoidToResult([dRow, dst, &reg, &copy]() -> void {
```

Comment:
I suggest to not use such constructs in performance-critical code.
While this code looks a bit nicer than when using a plain try...catch, there may be significant overhead introduced by creating function objects and dispatching to them. Apart from that, the captured variables will almost certainly lead to extra memory allocations for the lambda object. 
And as we do this per row, this code may be called quite often. So I strongly suggest changing this back to try...catch.